packages/auth/lib/Fields.js

```diff
@@ -0,0 +1,70 @@
+const moment = require('moment')
+
+const { newAuthError } = require('./AuthError')
+const Roles = require('./Roles')
+
+const MissingFieldsError = newAuthError(
+  'missing-fields',
+  'api/fields/missing'
+)
+
+const getMissingFields = async ({ user, email, pgdb }) => {
+  const missingFields = []
+  const unassignedGrants = await pgdb.public.accessGrants.find({
+    email,
+    recipientUserId: null,
+    'beginAt <=': moment(),
+    'endAt >': moment(),
+    invalidatedAt: null
+  })
+
+  const isMember = !!user && Roles.userHasRole(user, 'member')
+  const hasNames = !!user && (
```

Comment:
if you do this check first you could return early without doing the grant db query if one already has a name.